src/pages/ProductDetail.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import style from './ProductDetail.module.css';
+import { getProductDetail } from '../data/API';
+import { useState, useEffect } from 'react';
+import Button from '../components/ui/button/Button';
+import { addOrUpdateToCart } from '../api/firebase';
+import { useNavigate } from 'react-router-dom';
+import { useRecoilState } from 'recoil';
+import { loginState, userInfoState } from '../data/LoginData';
+import { BsCartX } from 'react-icons/bs';
+
+const SHIPPING = 3000;
+export default function ProductDetail() {
+  const [isLoggedIn, setIsLoggedIn] = useRecoilState(loginState);
+  const [userInfo, setUserInfo] = useRecoilState(userInfoState);
+  const userName = userInfo.displayName;
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const [detail, setDetail] = useState([]);
+  const { id: productId, title, photo, price, description, isSoldOut } = detail;
+
+  useEffect(() => {
+    const details = getProductDetail(id);
+    details.then((data) => {
+      setDetail(data);
+    });
+  }, [id]);
+
+  //최근 본 상품 로컬스토리지 저장 용도
+  useEffect(() => {
+    if (localStorage.watched === undefined) {
+      localStorage.setItem('watched', JSON.stringify([]));
+    }
+
+    const watched = localStorage.getItem('watched');
+    const watchedArr = JSON.parse(watched);
+    watchedArr.unshift(id);
+    localStorage.setItem('watched', JSON.stringify(watchedArr));
+
+    const watchedSet = new Set(watchedArr);
+    const watchedProducts = [...watchedSet].slice(0, 8);
+    localStorage.setItem('watched', JSON.stringify(watchedProducts));
+  }, []);
+
+  const handleClickCart = (e) => {
+    if (!isLoggedIn) {
+      const result = window.confirm('로그인이 필요한 서비스입니다. 로그인 하시겠습니까?');
+      if (result === true) window.location = '/login';
```

Comment:
```suggestion
  const handleClickCart = (event) => {
    if (!isLoggedIn) {
      const result = window.confirm('로그인이 필요한 서비스입니다. 로그인 하시겠습니까?');
      if (result) window.location = '/login';
```
`result===true` 대신 result 만 써도 될거같습니다.
event 는 e로 축약하지 않기로 합니다. [컨벤션](https://www.notion.so/hnhyerim/code-updated-12-22-f4f8aa0f6de9424ab2cad142df625d83#3a9a7b61ea1544f4b1ff6624c40e2894)